docs/best-practices/api-implementation.md

```diff
@@ -40,11 +40,13 @@ The code that implements these requests should not impose any side-effects. The
 
 If a POST request is intended to create a new resource, the effects of the request should be limited to the new resource (and possibly any directly related resources if there is some sort of linkage involved). For example, in an e-commerce system, a POST request that creates a new order for a customer might also amend inventory levels and generate billing information, but it should not modify information not directly related to the order or have any other side-effects on the overall state of the system.
 
-### Avoid implementing chatty POST, PUT, and DELETE operations
+### Batching
 
-Support POST, PUT, and DELETE requests over resource collections. A POST request can contain the details for multiple new resources and add them all to the same collection, a PUT request can replace the entire set of resources in a collection, and a DELETE request can remove an entire collection.
+Current computers are optimized for the repeated execution of the same task. If possible, you should structure your APIs and your program such that its execution alternates less frequently between different tasks. Batching can help you to achieve that and also it helps you to share out the cost of expensive operations such as I/O or expensive computations. For example, packing multiple data items into the same network packet or disk block to increase efficiency and reduce utilization.
 
-The Open Data Protocol (OData) support included in ASP.NET Web API 2 provides the ability to batch requests. A client application can package up several web API requests and send them to the server in a single HTTP request, and receive a single HTTP response that contains the replies to each request. For more information, see [Introducing batch support in Web API and Web API OData](https://blogs.msdn.microsoft.com/webdev/2013/11/01/introducing-batch-support-in-web-api-and-web-api-odata).
+Avoid implementing chatty operations. Whenever possible, provide API support for resource collections instead of just individual resources. A GET request can retrieve multiple resources at the same time to minimize the chance of overwhelming the application, a POST request can contain the details for multiple new resources and add them all to the same collection, a PUT request can replace the entire set of resources in a collection, and a DELETE request can remove an entire collection.
```

Comment:
```suggestion
Avoid designing your API based on _chatty_ operations. Every request comes with protocol, network, and compute overhead. For example, executing 100 smaller requests instead of 1 batch larger request incurs added overhead in the client, on the network, and at the resource server. Whenever possible, provide HTTP verb support for resource collections instead of just individual resources.

- A GET request to a collection can retrieve multiple resources at the same time.
- A POST request can contain the details for multiple new resources and add them all to the same collection.
- A PUT request can replace the entire set of resources in a collection.
- A DELETE request can remove an entire collection.
```